### ISSUE TITLE
Errors with imfil for univariate functions

### ISSUE THREAD

**Issue by salperinlea** [scikit-quant/scikit-quant#2]
I'll be posting four different issues since, even though the code to reproduce them is nearly the same, the four cases throw different errors. 

To reproduce the first error
```
import numpy as np
from skquant.opt import minimize

def f(a):
    return a**2 - a

bounds=np.array([[0,2]],dtype=np.float)
init=np.array([1.])
res,hist=minimize(f,init,bounds,method='imfil')
```

Error is:

```
Traceback (most recent call last):
  File "/Users/sumneralperin-lea/science_git/Tequila/examples/debug.py", line 9, in <module>
    res,hist=minimize(f,init,bounds,method='imfil')
  File "/Users/sumneralperin-lea/git/scikit-quant/skquant/opt/__init__.py", line 54, in minimize
    return optimizer.minimize(func, x0, bounds, budget, options, **optkwds)
  File "/Users/sumneralperin-lea/anaconda3/envs/quantum/lib/python3.6/site-packages/SQImFil/_imfil.py", line 208, in minimize
    imfil_core(z0, f_internal, budget, core_data, bounds)
  File "/Users/sumneralperin-lea/anaconda3/envs/quantum/lib/python3.6/site-packages/SQImFil/_imfil.py", line 391, in imfil_core
    funs, iff, icf, noise_val = f(x, h, core_data)
  File "/Users/sumneralperin-lea/anaconda3/envs/quantum/lib/python3.6/site-packages/SQImFil/_imfil.py", line 591, in f_internal
    res = objfunc(*call_args)
  File "/Users/sumneralperin-lea/anaconda3/envs/quantum/lib/python3.6/site-packages/SQCommon/_objective.py", line 39, in __call__
    self.stats.add_history(val, par)
  File "/Users/sumneralperin-lea/anaconda3/envs/quantum/lib/python3.6/site-packages/SQCommon/_stats.py", line 14, in add_history
    self._history.append([fval]+list(par))
TypeError: iteration over a 0-d array

```

**Comment by salperinlea** [comment]
Error NOT fixed by upgrade to sqimfil 0.3.4; remains the same.

**Comment by wlav** [comment]
Yes; Matthias' fix does the trick. Regression test has been added.
